### PR TITLE
shellcheck doc-index.sh & apply suggestions

### PR DESCRIPTION
**Pull request by Krakn**
Bring script in line with Google's shell style suggestions:
 - https://google.github.io/styleguide/shell

Changelog:
 - 'set -u': Treat unset variables as errors
 - Global variables are now readonly
 - Move functions into main function entrypoint

Part of: #114

### CODE REVIEW

File: ci/doc-index.sh [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 
-DOC_INDEX_PAGE_URL=${DOC_INDEX_PAGE_URL:-https://neovim.io/doc_index}
+source "${BUILD_DIR}/ci/common/common.sh"
+source "${BUILD_DIR}/ci/common/doc.sh"
 
 generate_doc_index() {
-  echo "Updating index.html from ${DOC_INDEX_PAGE_URL}."
-  curl -L --tlsv1 ${DOC_INDEX_PAGE_URL} > ${DOC_DIR}/index.html
+  echo "Updating index.html from: ${DOC_INDEX_PAGE_URL}"
+  # TODO: Can the 'tlsv1' flag be removed?
```
Comment (justinmk) [inline ci/doc-index.sh]: see 0e9f559802ccc53530a4b2d3afab703f49df8e19

So, don't need that comment. If it is proven that it isn't needed we can remove it later.

**Comment by Krakn** [comment]
@justinmk I removed the comment.

This should now be ready to merge.

File: ci/doc-index.sh [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
-  echo "Updating index.html from ${DOC_INDEX_PAGE_URL}."
-  curl -L --tlsv1 ${DOC_INDEX_PAGE_URL} > ${DOC_DIR}/index.html
+  echo "Updating index.html from: ${DOC_INDEX_PAGE_URL}"
+  curl -L --tlsv1 "${DOC_INDEX_PAGE_URL}" -o "${DOC_DIR}/${DOC_SUBTREE}"
+}
+
+main() {
+  clone_doc
+  generate_doc_index
+  #commit_doc
```
Comment (justinmk) [inline ci/doc-index.sh]: why is this commented out?

Comment (Krakn) [inline ci/doc-index.sh]: It was commented out from my personal testing branch. Good catch.

**Pull request merged**